avantifellows/portal-frontend
# Auto redirection to destination if the token is valid

If a student has already logged in and a token is stored, then the token is verified for authentication of user and authorization of the group the user belongs to. If its valid, then the user is redirected to the destination rather than making them sign-in again.

**Types of links to test:**

IDs:
Haryana : 1509117156
 
Delhi : 20210270337 .. 18-07-2007
 
Tnschools : UDM301
 
Punjabteachers : 11033090508103

 
Simple haryana session: http://staging-auth.avantifellows.org/?sessionId=HaryanaStudents_HaryanaStudents_10_Foundation_24_001,HaryanaStudents_9_Foundation_24_001_45511_1mfzzeg2cbk-pj1w6wosj6qskulo12xwy
 
Gurukul: http://staging-auth.avantifellows.org/?group=DelhiStudents&platform=gurukul (delhi)
 
Reporting: http://staging-auth.avantifellows.org/?type=attendance&redirectTo=report&redirectId=DelhiStudents_66bb77be0beaf750e503f560&sub_type=reporting&group=DelhiStudents&auth_type=ID,DOB (delhi)

Delhi Liveclass: http://staging-auth.avantifellows.org/?sessionId=DelhiStudents_DelhiStudents_11_Photon_Eng_23_001,DelhiStudents_12_Photon_med_24_S001_45524_zvp-wyjs-whd

PunjabTeachers (v1): http://staging-auth.avantifellows.org/?sessionId=PunjabTeachers_12_med_004_24_45421_riFbP4lEeNc
 
Registration id generation: http://staging-auth.avantifellows.org/?sessionId=UttarakhandStudents_Registration_11_24_45498
 
Registration without id generation: http://staging-auth.avantifellows.org/?sessionId=DelhiStudents_Registration_12_24_45498
 
Tnschools: http://staging-auth.avantifellows.org/?sessionId=TNSchools_TNStudents_12_Photon_Eng_24_B001_TNStudents_12_Photon_med_24_B001_45489_84024064624

**How to test?**
1. Login to gurukul (this saves token)
2. When you access sessions from gurukul, you should not see login screen. Auto redirection.
3. When you access "Delhi" sessions, auto redirect again.
4. When you access "Non-Delhi" sessions, you should see login screen. 
5. When you logout from gurukul and access "Delhi" sessions, you should see login screen (no redirection).
6. V1 links should work seamlessly.
7. To test refresh token after access expiry, wait for an hour and then open "Delhi" sessions.
